src/netbase.cpp

```diff
@@ -632,11 +632,6 @@ std::unique_ptr<Sock> ConnectDirectly(const CService& dest, bool manual_connecti
 
 std::unique_ptr<Sock> Proxy::Connect() const
 {
-    if (!IsValid()) {
```

Comment:
It's probably a good idea to keep the `!IsValid` check here, just to be sure. Fine with removing the log.

Reply:
I could be mistaken, but it looks at first glance that this would be covered in `ConnectToSocket()`:

https://github.com/bitcoin/bitcoin/blob/d1af4422d13ba84253a1555c2fe5a42170fa8b35/src/netbase.cpp#L603

and reached in Proxy::Connect() through either the call to `ConnectToSocket()`:
https://github.com/bitcoin/bitcoin/blob/d1af4422d13ba84253a1555c2fe5a42170fa8b35/src/netbase.cpp#L653

or by way of `ConnectDirectly()`:
https://github.com/bitcoin/bitcoin/blob/d1af4422d13ba84253a1555c2fe5a42170fa8b35/src/netbase.cpp#L626

Reply:
Yes, it's not strictly currently necessary, but code changes sometimes. It's good to have error and unexpected situation checking in place, instead of crashing later on.